filters/all.js

```diff
@@ -7,15 +7,52 @@ const { AsyncAPIDocument } = require('@asyncapi/parser');
 
 const filter = module.exports;
 
+function isJsonObject(o) {
+  return o && typeof o === 'object' && !Array.isArray(o);
+}
+
+/**
+ * Performs a recursive deep merge while assuming only simple JSON types are used.
+ */
+function mergeInto(from, to) {
+  for (const key in from) {
+    if (!Object.prototype.hasOwnProperty.call(from, key)) {
+      continue;
+    }
+    if (isJsonObject(from[key])) {
+      if (!isJsonObject(to[key])) {
+        to[key] = {};
+      }
+      mergeInto(from[key], to[key]);
+    } else {
+      // Override with non-object JSON value
+      to[key] = from[key];
+    }
+  }
+}
+
 /**
  * Prepares configuration for component.
  */
 function prepareConfiguration(params = {}) {
   const config = { show: { sidebar: true }, sidebar: { showOperations: 'byDefault' } };
+  // Apply config override
+  if (params.config) {
+    let configOverride;
+    try {
+      configOverride = JSON.parse(params.config);
+    } catch (err) {
+      // Failed to parse JSON, attempt to read as JSON file
+      configOverride = JSON.parse(fs.readFileSync(params.config, "utf8"));
+    }
```

Comment:
From what I can see you will be operating on an absolute path and not a local path. It would be nice to fix this somehow to give the option of two options of local path and absolute path - I'm not sure how this `readFile` function works underneath, if it tries to check the absolute path and then the local path in case of an error. Can you enlighten me? We should first check for absolute path and then check local using `cwd`.

Also it would be nice to make some `console.error` before throwing error that given config isn't valid JSON/file path.

Reply:
All relative paths are resolved against the current working directory in the Node.js `fs` module. The code above first tries to parse the config as stringified json, then falls back to assuming the config string is a file path and attempts to read it. I agree that we should add a second try catch around the file loading and report a friendly error message. Should it be a fatal error or just print and continue?

Reply:
Updated. Let me know if the second try-catch is sufficient to provide a friendly error message with `console.error`. Also added more comments to explain what the code does.

Reply:
> All relative paths are resolved against the current working directory in the Node.js fs module. The code above first tries to parse the config as stringified json, then falls back to assuming the config string is a file path and attempts to read it. I agree that we should add a second try catch around the file loading and report a friendly error message. Should it be a fatal error or just print and continue?

Thanks for clarification! Something light in my head regarding the default cwd but I couldn't quite remember.

> Updated. Let me know if the second try-catch is sufficient to provide a friendly error message with console.error. Also added more comments to explain what the code does.

Thanks! `console.error` is sufficient, but I think that after logging we should throw error (`err` or `jsonErr` but `err` will be better) to stop executing rest of the code - generator should stop there and not continue generation.

Reply:
Done.